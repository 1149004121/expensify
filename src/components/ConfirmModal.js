import React from 'react';
import Modal from 'react-modal';

export default class ConfirmModal extends React.Component {
  render() {
    return (
      <Modal
        isOpen={!!this.props.option}
        className="modal">
        <h3>Are you sure you want to delete?</h3>
        <div className='choices'>
          <button onClick={this.props.deleteItem} className="confirm">Yes</button>
          <button onClick={this.props.reserveItem} className="reserve">No</button>
        </div>
      </Modal>
    )
  }
}