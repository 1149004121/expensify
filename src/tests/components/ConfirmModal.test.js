import ConfirmModal from '../../components/ConfirmModal';
import React from 'react';
import { shallow } from 'enzyme';

let deleteItem, reserveItem;

beforeEach(() => {
  deleteItem = jest.fn();
  reserveItem = jest.fn();
})

test('should hide ConfirmModal', () => {
  const wrapper = shallow(<ConfirmModal
    deleteItem={deleteItem}
    reserveItem={reserveItem}
    option={undefined}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should show ConfirmModal', () => {
  const wrapper = shallow(<ConfirmModal
    deleteItem={deleteItem}
    reserveItem={reserveItem}
    option={true}
  />);
  expect(wrapper).toMatchSnapshot();
});



