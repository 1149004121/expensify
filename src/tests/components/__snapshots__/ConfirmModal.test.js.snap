// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide ConfirmModal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3>
    Are you sure you want to delete?
  </h3>
  <div
    className="choices"
  >
    <button
      className="confirm"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="reserve"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;

exports[`should show ConfirmModal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3>
    Are you sure you want to delete?
  </h3>
  <div
    className="choices"
  >
    <button
      className="confirm"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="reserve"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;
